probe_scraper/runner.py

```diff
@@ -102,19 +120,84 @@ def main(temp_dir, out_dir):
                                                           break_by_channel=False)
 
     # Serialize the probe data to disk.
-    write_probe_data(probes_by_channel, revisions, out_dir)
+    write_moz_central_probe_data(probes_by_channel, revisions, out_dir)
+
+
+def load_http_probes(date, cache_dir="cache", out_dir="output"):
+    repos_probes_data, emails = http_scraper.scrape(date, cache_dir)
+
+    # Parse probe data from files into the form:
+    # <repo_name> -> {
+    #   <date> -> {
+    #     "histogram": {
+    #       <histogram_name>: {
+    #         ...
+    #       },
+    #       ...
+    #     },
+    #     "scalar": {
+    #       ...
+    #     },
+    #   },
+    #   ...
+    # }
+    probes = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
+    for repo_name, dates in repos_probes_data.iteritems():
+        for date, probe_types in dates.iteritems():
+            for probe_type, paths in probe_types.iteritems():
+                try:
+                    results = PARSERS[probe_type].parse(paths)
+                except Exception:
+                    msg = "Improper file in {}\n{}".format(', '.join(paths), traceback.format_exc())
+                    emails[repo_name]["emails"].append({
+                        "subject": "Probe Scraper: Improper File",
+                        "message": msg
+                    })
+                probes[repo_name][date][probe_type] = results
+
+    probes_by_repo = transform_probes.transform_by_date(probes)
+
+    write_external_probe_data(probes_by_repo, out_dir)
+
+    for repo_name, email_info in emails.items():
+        addresses = email_info["addresses"] + [DEFAULT_TO_EMAIL]
+        for email in email_info["emails"]:
+            send_ses(FROM_EMAIL, email["subject"], email["message"], addresses)
+
+
+def main(date, cache_dir, out_dir, process_moz_central, process_http):
+    if process_moz_central:
+        load_moz_central_probes(cache_dir, out_dir)
+    if process_http:
+        load_http_probes(date, cache_dir, out_dir)
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument('--tempdir',
-                        help='Temporary directory to work in. This serves as a cache if reused.',
+    parser.add_argument('--date',
+                        help='The date being run',
+                        action='store',
+                        required=True)
+    parser.add_argument('--cache-dir',
+                        help='Cache directory. If empty, will be filled with the probe files.',
                         action='store',
                         default=tempfile.mkdtemp())
-    parser.add_argument('--outdir',
+    parser.add_argument('--out-dir',
                         help='Directory to store output files in.',
                         action='store',
                         default='.')
+    parser.add_argument('--skip-moz-central-probes',
```

Comment:
This might be easier to understand as `--only-moz-central-probes` or similar.
This is not a blocker if that's involved to change.